pkg/kubelet/userns/userns_manager.go

```diff
@@ -157,7 +157,7 @@ func MakeUserNsManager(kl userNsPodsManager) (*UsernsManager, error) {
 	for _, podUID := range found {
 		klog.V(5).InfoS("reading pod from disk for user namespace", "podUID", podUID)
 		if err := m.recordPodMappings(podUID); err != nil {
-			return nil, err
+			return nil, fmt.Errorf("user namespace manager can't record pod mappings: %w", err)
```

Comment:
Would it be better to do this on the lower level, considering that it's already partly done: `recordPodMappings` calls `parseUserNsFileAndRecord`, which returns properly formatted errors.

Reply:
I don't think so. `parseUserNsFileAndRecord()` is not called unconditionally, it can fail before calling that. I've updated the commit message with examples of how it failed before and how it fails now, so it's clear the improvement.

But I'll put them here too. The kubelet logs will show something like this now:
    
            E0120 16:02:40.484404  474711 run.go:74] "command failed" err="failed to run Kubelet: failed to create kubelet: user namespace manager can't record pod mappings: mkdir XXX: permission denied"
    
Before this commit, the message was not so clear:
    
            E0120 16:02:40.484404  474711 run.go:74] "command failed" err="failed to run Kubelet: failed to create kubelet: mkdir XXX: permission denied"
    
Which is not clear at all, the run.go:74 is in a very generic file that doesn't give you many clues of what is really calling.

Even for the error from `parseUserNsFileAndRecord()`, this is way more clear now.

Reply:
That's exactly what I mean: if **all** lower level functions called from recordPodMappings would return properly formatted errors, you wouldn't need to wrap them in this function. `parseUserNsFileAndRecord` does this, but it's not the only function called from `recordPodMappings`.

Reply:
But how would the error printed on kubelet initialization look like? I don't think it will be clear it is the user namespace manager that is failin to do something, right?

I'm open to ideas, please share how the example log would look like, my thinking was that it wasn't more clear than this (for this log line that is printed in a mega generic function, so the line doesn't give you any useful hint), that is why I did it. But maybe I missed something?

Reply:
Here is what I think about it. We have unwrapped error "mkdir: permission denied" produced by the code in the `pkg/kubelet/util/store/filestore.go:NewFileStore`. This function is called from 3 different places in the k/k code (even from 4 if we consider test code). I think it makes sense to wrap the error on the lowest possible level than wrapping it in 4 different places. However, if something like "NewFileStore: failed to create file store: %w" is not clear enough, we can decide to wrap it again at a higher level. Alternatively we can wrap it in all places where NewFileStore is called and remember to wrap it every time we use NewFileStore.

In my opinion "user namespace manager can't record pod mappings:  mkdir: permission denied" gives more hints than "NewFileStore: failed to create file store: mkdir permissions denied", but this is just my opinion.

Reply:
The big difference IMHO is that you can grep the userns error and you will find the issue quite easily, while it is not that simple if you only wrap it on the lowest level (it is called from different places, it is not clear where it is called during initialization, etc.). If we wrap it at the userns manager level, we can easily find the issue.

I'm not against adding more wrappers, though, if you have a specific suggestion (i.e. what message to add and where), I can add that too.

Reply:
@bart0sh friendly ping?

AFAIK, this is not waiting on author as it is marked now

Reply:
Sorry, missed your answer, but still tend to disagree that "user namespace manager can't record pod mappings: mkdir: permission denied" doesn't give more hints than "NewFileStore: failed to create file store: mkdir permissions denied". From my point of view latter is more logical and understandable.

I'd give other reviewers an opportunity to express their opinions.

Reply:
@bart0sh I added wrappings for file store too now (and simplified some other error messages along the way). Let me know what you think!

Reply:
/lgtm